src/server/stream_family.cc

```diff
@@ -1640,6 +1871,24 @@ void StreamFamily::XRead(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void StreamFamily::XRead(CmdArgList args, ConnectionContext* cntx) {
+  auto opts = ParseReadArgsOrReply(args, false, cntx);
+  if (!opts) {
+    return;
+  }
+
+  return XReadGeneric(args, opts, cntx);
+}
+
+void StreamFamily::XReadGroup(CmdArgList args, ConnectionContext* cntx) {
+  auto opts = ParseReadArgsOrReply(args, true, cntx);
+  if (!opts) {
+    return;
+  }
+
+  return XReadGeneric(args, opts, cntx);
+}
+
```

Comment:
These two functions only differ on the boolean flag `true/false` passed down to `ParseReadArgsOrReply`, extract the common logic into a separate function and call it from there (see suggestion below).